docs/Anjay_integrations/ESP32_integration/M5Stick.md

```diff
@@ -0,0 +1,132 @@
+# M5StickC
+
+Integrate your ESP32-based device to manage it via Coiote DM.
+
+## Prerequisites
+
+- An M5StickC device.
+- Installed ESP-IDF and dependencies (installation steps 1-4 from [ESP32 official documentation](https://docs.espressif.com/projects/esp-idf/en/v4.4/esp32/get-started/index.html)). Supported ESP-IDF version is v4.4.
+- A user with access to the Coiote IoT Device Management platform.
+
+## Step 1: Clone Anjay ESP32 client repository
+
+Enter the command line interface on your machine and paste the following command:
+
+   ```
+   git clone --recursive https://github.com/AVSystem/Anjay-esp32-client
+   ```
+
+## Step 2: Configure the client using an NVS partition
+
+0. Create a `nvs_config.csv` file and provide your credentials in [wifi_ssid], [wifi_password], [identity], [psk], [lwm2m_server_uri] (without the `[]` brackets). You can use the following snippet as a template:
+
+    !!! important
+        The **Identity** parameter stands for both the device endpoint name and its PSK identity, therefore these two must be identical in Coiote DM.  
+
+    ```
+    key,type,encoding,value
+    config,namespace,,
+    wifi_ssid,data,string,[wifi_ssid]
+    wifi_pswd,data,string,[wifi_password]
+    wifi_inter_en,data,u8,1
+    identity,data,string,[identity]
+    psk,data,string,[psk]
+    uri,data,string,[lwm2m_server_uri]
+    writable_wifi,namespace,,
+    wifi_ssid,data,string,[wifi_ssid]
+    wifi_pswd,data,string,[wifi_password]
+    wifi_inter_en,data,u8,0
+    ```
+
+    !!! Note
+        The additional parameters under the **writable_wifi** namespace are used to provide a secondary Wi-Fi configuration (it is not obligatory). This allows for switching between Wi-Fi configurations while the device is running.
+
+0. Generate the NVS partition:
+
+```
+python3 tools/nvs_partition_gen.py generate nvs_config.csv nvs_config.bin 0x4000
+```
+
+![Client configuration](images/nvs_config.png "Client configuration"){: style="float: left;margin-right: 1177px;margin-top: 17px;"}
+
+## Step 4: Add device to Coiote DM
+
+To connect your M5StickC to the Coiote IoT Device Management LwM2M Server, use your access to a Coiote DM installation, or register at https://www.avsystem.com/try-anjay/ to get access.
+
+To connect the board:
+
+0. Log in to Coiote DM and from the left side menu, select **Device Inventory**.
+0. In **Device Inventory**, click **Add device**.
+0. Select the **Connect your LwM2M device directly via the Management server** tile.
+![Add via Mgmt](images/mgmt_tile.png "Add via Mgmt")
+0. In the **Device credentials** step:
+    - In the **Device ID** field, type the name provided in the `nvs_config.csv`, e.g. `ESP32_test`.
```

Comment:
Maybe we shouldn't divide this subpoints with image?

Reply:
ok